nimble/host/src/ble_hs_hci.c

```diff
@@ -643,3 +643,11 @@ ble_hs_hci_init(void)
                             "ble_hs_hci_frag");
     BLE_HS_DBG_ASSERT_EVAL(rc == 0);
 }
+
+void
+ble_hs_hci_deinit(void)
+{
+    ble_npl_mutex_deinit(&ble_hs_hci_mutex);
+
+    ble_npl_sem_deinit(&ble_hs_hci_sem);
+}
```

Comment:
Should this function (`ble_hs_hci_deinit`) also include:
```
ble_npl_eventq_deinit(&ble_hci_sock_state.evq);
```

?

Reply:
If I am not wrong, then it is used for socket transport. If you see, `ble_hci_sock_init` in which the eventq is inited is called from the application and not from the host stack. So, ideally its corresponding deinit function should also be a part of a separate API like `ble_hci_sock_deinit` which should be called by the application (only for socket transport). Also, `ble_hs_hci_deinit` is generic and required for all transports. 

I had verified with my PR that there were no leaks between start and stop (we do not use socket transport)